extensions/positron-python/python_files/posit/positron/inspectors.py

```diff
@@ -1206,3 +1250,116 @@ def _get_kind(value: Any) -> str:
+    try:
+        display_value = repr(value)
+    except Exception:
+        try:
+            display_value = object.__repr__(value)
+        except Exception:
+            try:
+                display_value = f"<no repr available for {get_qualname(value)}>"
+            except Exception:
+                display_value = "<no repr available for object>"
+
+    return _maybe_truncate_string(display_value, level=level)
+
+
+def _get_string_display_value(
+    value: str | bytes | bytearray, *, level: int = 0
+) -> tuple[str, bool]:
+    # Limit the length of the displayed string; the limit depends on whether the string
+    # is nested in another object.
+    max_characters = MAX_CHARACTERS_NESTED if level > 0 else MAX_CHARACTERS
+
+    # If the string fits the limit, return it as is.
+    # NOTE: We check the actual string length but return the repr which may differ e.g.
+    #       some characters may be escaped. We don't check `len(repr(value))` to avoid
+    #       possibly copying a huge string.
+    if len(value) <= max_characters:
+        return repr(value), False
+
+    # Split the character budget into two parts: 2/3 for the start and 1/3 for the end.
+    start_characters = max(1, int(2 * max_characters / 3))
+    end_characters = max(1, int(max_characters / 3))
+
+    # Only compute the repr after slicing to avoid duplicating large strings
+    # and remove surrounding quotes (and string prefixes).
+    start = repr(value[:start_characters])
+    start = start[: start.rindex("'")]
+    end = repr(value[-end_characters:])
+    end = end[end.index("'") + 1 :]
+    return f"{start}{ELLIPSIS}{end}", True
+
+
+def _get_collection_display_value(
+    value: CollectionT | Column,
+    *,
+    prefix: str,
+    suffix: str,
+    level: int = 0,
+):
+    # If the collection is empty, return the empty representation.
+    if len(value) == 0:
+        return f"{prefix}{suffix}", False
+
+    # If we're at the max nested level, just return ellipsis e.g. `[...]`.
+    if level >= len(MAX_ITEMS_BY_LEVEL):
+        return f"{prefix}{ELLIPSIS}{suffix}", True
+
+    # Loop through the items in the collection and build the display value.
+    parts = [prefix]
+    item_allowance = MAX_ITEMS_BY_LEVEL[level]
+    should_add_separator = False
+    truncated = False
+    for item in value:
+        # Add a separator, if needed.
+        if should_add_separator:
+            parts.append(", ")
+        should_add_separator = True
+
+        # If we've reached the item allowance, add ellipsis and break.
+        item_allowance -= 1
+        if item_allowance <= 0:
+            parts.append(ELLIPSIS)
+            truncated = True
+            break
+
+        # Get the display value for the item.
+        #
+        # For items that are the same object as the parent, avoid infinite recursion
+        # and represent the item with ellipsis.
+        if item is value:
+            item_repr, item_truncated = f"{prefix}{ELLIPSIS}{suffix}", True
```

Comment:
<img width="527" height="361" alt="Screenshot 2025-07-17 at 10 15 52 AM" src="https://github.com/user-attachments/assets/67613079-b828-4971-97b4-11b896fefc2e" />

not needed for this PR, but this can get you in a weird state with double ellipses if you end up resizing the variables pane

Reply:
Good catch. It does look like an existing issue on `main`. It isn't technically wrong but I agree it's not ideal. I also feel like I want to double-click it to expand the display value somehow.

<img width="487" height="137" alt="image" src="https://github.com/user-attachments/assets/bd6ff5bb-7371-449d-b22a-791ba6af2c8f" />